contk/dataloader/multi_turn_dialog.py

```diff
@@ -288,3 +291,391 @@ def _load_data(self):
+				origin_data['dialog_acts'].append([None] + \
+													   list(map(lambda utt: utt[2][0], line['utts'])))
+				origin_data['session'].append(sess_prefix + list(map(lambda utt: \
+														utt[1].strip().split(), line['utts'])))
+		return origin_data
+
+	def _build_vocab(self, origin_data):
+		r'''Building vocabulary(words, topics, da), invoked by `SwitchboardCorpus._load_data`
+		'''
+		vocab = list(chain(*chain(*origin_data['train']['session'])))
+		vocab = sorted(Counter(vocab).most_common(), key=lambda pair: (-pair[1], pair[0]))
+		left_vocab = vocab[:self._max_vocab_size]
+		left_vocab = list(map(lambda x: x[0], left_vocab))
+		if '<eos>' in left_vocab:
+			left_vocab.remove('<eos>')
+		vocab_list = self.ext_vocab + left_vocab
+		self.word2id = {w: i for i, w in enumerate(vocab_list)}
+
+		self.topic_vocab_list = sorted(list(set(origin_data['train']['topics'])))
+		self.topic2id = {t: i for i, t in enumerate(self.topic_vocab_list)}
+		self.da_vocab_list = list(set(list(chain(*origin_data['train']['dialog_acts']))))
+		self.da_vocab_list.remove(None)
+		self.da_vocab_list = sorted(self.da_vocab_list)
+		self.da2id = {a: i for i, a in enumerate(self.da_vocab_list)}
+
+		print("vocab list length = %d" % len(vocab_list))
+		print("topic vocab list length = %d" % len(self.topic_vocab_list))
+		print("dialog act vocab list length = %d" % len(self.da_vocab_list))
+		return vocab_list
+
+	def _load_data(self):
+		r'''Loading dataset, invoked by `MultiTurnDialog.__init__`
+		'''
+		origin_data = {}
+		for key in self.key_name:
+			origin_data[key] = self._read_file('%s/switchboard_corpus_%s.jsonl' % (self._file_path, key))
+
+		vocab_list = self._build_vocab(origin_data)
+
+		data = {}
+		data_size = {s: 0 for s in self.key_name}
+		for key in self.key_name:
+			data[key] = self._convert2ids(origin_data[key])
+			data_size[key] = len(data[key]['session'])
+
+			vocab = list(chain(*chain(*origin_data[key]['session'])))
+			vocab_num = len(vocab)
+			oov_num = len(list(filter(lambda word: word not in self.word2id, vocab)))
+			sent_length = list(map(len, chain(*origin_data[key]['session'])))
+			cut_word_num = float(np.sum(np.maximum(np.array(sent_length) - self._max_sen_length + 2, 0)))
+			print("%s set. OOV rate: %f, max length before cut: %d, cut word rate: %f" % \
+				  (key, oov_num / vocab_num, max(sent_length), cut_word_num / vocab_num))
+		if os.path.exists('%s/switchboard_corpus_multi_ref.jsonl' % self._file_path):
+			self.key_name.append('multi_ref')
+			data['multi_ref'] = self._load_multi_ref_data()
+			data_size['multi_ref'] = len(data['multi_ref']['session'])
+		return vocab_list, len(vocab_list), data, data_size
+
+	def get_batch(self, key, index):
+		'''Get a batch of specified `index`.
+
+		Arguments:
+			key (str): must be contained in `key_name`
+			index (list): a list of specified index
+
+		Returns:
+			(dict): A dict at least contains:
+
+				* session (dict): Return of MultiTurnDialog.get_batch
```

Comment:
please don't make another dict there, just let the contents in `session` be items of `res`